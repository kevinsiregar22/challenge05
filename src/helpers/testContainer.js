import React from 'react';
import {Provider} from 'react-redux';
import {store} from '../store';

export default function testContainer(component) {
  return <Provider store={store}>{component}</Provider>;
}
