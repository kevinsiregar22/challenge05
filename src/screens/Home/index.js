import {StyleSheet, View, TouchableOpacity, FlatList} from 'react-native';
import axios from 'axios';
import {colors} from '../../utils';
import {API} from '../../config/API';
import React, {useEffect} from 'react';
import {SetRecommend} from './redux/action';
import {ms} from 'react-native-size-matters';
import Loading from '../../components/Loading';
import Poppins from '../../components/Poppins';
import FastImage from 'react-native-fast-image';
import {navigate} from '../../helpers/navigate';
import {setLoading} from '../../store/globalAction';
import {useDispatch, useSelector} from 'react-redux';

const Index = () => {
  const dispatch = useDispatch();

  const {recommend} = useSelector(state => state.home);
  const {isLoading} = useSelector(state => state.Global);
  const {token, name} = useSelector(state => state.login);

  useEffect(() => {
    recommended();
  }, []);

  const recommended = async () => {
    try {
      dispatch(setLoading(true));

      const res = await axios.get(API.BASE_API.concat('/books'), {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      console.log(res.data);
      console.log(res.data.results, 'result');
      dispatch(SetRecommend(res.data.results));
    } catch (error) {
      console.log(error);
    } finally {
      dispatch(setLoading(false));
    }
  };

  if (isLoading) {
    return <Loading />;
  }

  return (
    <View style={styles.container}>
      <Poppins textAlign="left" marginBottom={5} marginTop={4}>
        Selamat Datang, {name}
      </Poppins>
      <Poppins type="Bold" size={20} textAlign="left">
        Recommenhded
      </Poppins>
      <FlatList
        keyExtractor={item => item.id}
        data={recommend}
        renderItem={({item}) => (
          <TouchableOpacity
            onPress={() => navigate('DetailBooks', {id: item.id})}>
            <View style={styles.romended}>
              <FastImage
                style={styles.imageRecomen}
                source={{uri: item.cover_image}}
                resizeMode={FastImage.resizeMode.cover}
              />
            </View>
          </TouchableOpacity>
        )}
        horizontal={true}
      />
      <Poppins type="Bold" size={30} textAlign="left">
        Popular Book
      </Poppins>
      <FlatList
        keyExtractor={item => item.id}
        numColumns={3}
        data={recommend}
        horizontal={false}
        renderItem={({item}) => (
          <TouchableOpacity
            onPress={() => navigate('DetailBooks', {id: item.id})}>
            <View style={styles.popular}>
              <FastImage
                style={styles.imagesPopular}
                source={{uri: item.cover_image}}
                resizeMode={FastImage.resizeMode.cover}
              />
              <Poppins textAlign="left" marginLeft={2}>
                {item.title}
              </Poppins>
            </View>
          </TouchableOpacity>
        )}
      />
    </View>
  );
};

export default Index;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.backgroundPage,
  },

  romended: {
    left: ms(8),
    margin: ms(5),
    width: ms(130),
    height: ms(320),
    borderRadius: 10,
  },

  imageRecomen: {
    width: ms(130),
    height: ms(180),
    borderRadius: 10,
  },

  popular: {
    left: 10,
    margin: 5,
    width: ms(115),
    height: ms(240),
    borderRadius: 10,
    marginBottom: ms(20),
    backgroundColor: colors.cart.color,
  },

  imagesPopular: {
    width: ms(115),
    height: ms(160),
    borderRadius: 10,
    justifyContent: 'space-between',
  },
});
