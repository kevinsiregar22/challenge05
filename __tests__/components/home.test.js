import React from 'react';
import {create} from 'react-test-renderer';
import Home from '../..//src/screens/Home';
import testContainer from '../../src/helpers/testContainer';

describe('should loading', () => {
  test('should first loading', () => {
    const component = create(testContainer(<Home />)).toJSON();
    expect(component).toMatchSnapshot();
  });
});
