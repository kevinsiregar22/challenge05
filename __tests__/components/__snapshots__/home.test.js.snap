// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should loading should first loading 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D5EFCF",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
        "marginBottom": 5,
        "marginLeft": 15,
        "marginRight": 1,
        "marginTop": 4,
        "textAlign": "left",
      }
    }
    testID="text component"
  >
    Selamat Datang, 
    
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "Poppins-Bold",
        "fontSize": 20,
        "marginBottom": 10,
        "marginLeft": 15,
        "marginRight": 1,
        "marginTop": 1,
        "textAlign": "left",
      }
    }
    testID="text component"
  >
    Recommenhded
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {},
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": 10,
                "height": 502.85714285714283,
                "left": 12.571428571428571,
                "margin": 7.857142857142857,
                "width": 204.28571428571428,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 10,
                    "height": 282.8571428571429,
                    "width": 204.28571428571428,
                  },
                ]
              }
            >
              <FastImageView
                resizeMode="cover"
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "Poppins-Bold",
        "fontSize": 30,
        "marginBottom": 10,
        "marginLeft": 15,
        "marginRight": 1,
        "marginTop": 1,
        "textAlign": "left",
      }
    }
    testID="text component"
  >
    Popular Book
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {},
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#D6EAFF",
                  "borderRadius": 10,
                  "height": 377.1428571428571,
                  "left": 10,
                  "margin": 5,
                  "marginBottom": 31.428571428571427,
                  "width": 180.71428571428572,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 10,
                      "height": 251.42857142857142,
                      "justifyContent": "space-between",
                      "width": 180.71428571428572,
                    },
                  ]
                }
              >
                <FastImageView
                  resizeMode="cover"
                  source={
                    Object {
                      "uri": undefined,
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 16,
                    "marginBottom": 10,
                    "marginLeft": 2,
                    "marginRight": 1,
                    "marginTop": 1,
                    "textAlign": "left",
                  }
                }
                testID="text component"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
